### PR TITLE
Update robolectric and mockito to fix unit tests

### PR DESCRIPTION
**Pull request by JonasVautherin**
Should fix the remaining tests.

* All but one were fixed by updating roboelectric and mockito (so this is a safe change)
* One was failing on a call I don't understand, that internally calls some geoJson stuff (which I removed in many places in a previous PR because 20 tests were failing too, for a reason I also don't understand). Honestly I don't understand it, I just removed as little as possible to make the tests pass.
* After all the previous tests passed, Gradle moves on to building more tests, and one of them failed to build. I fixed that, too.

EDIT: now it passes :tada: 

@fynngodau @louwers: happy to get a review :blush:

### CODE REVIEW

File: plugin-annotation/src/test/java/com/mapbox/mapboxsdk/plugins/annotation/DraggableAnnotationControllerTest.java [hunk 1]
```diff
@@ -265,7 +265,6 @@ public void gestureOnMoveTest() {
 
         assertTrue(moved);
         verify(annotation).setGeometry(geometry);
-        verify(annotationManager).internalUpdateSource();
```
Comment (fynngodau) [inline plugin-annotation/src/test/java/com/mapbox/mapboxsdk/plugins/annotation/DraggableAnnotationControllerTest.java:268]: The reason is my PR that delays calls to `internalUpdateSource` using `postUpdateSource`, to group multiple calls in immediate succession from the client into only a single O(n) call: https://github.com/maplibre/maplibre-plugins-android/pull/12

It _should_ work to verify that `postUpdateSource` is called instead. (Tests aren't running for me locally so it's hard to say for sure.)

**Review by fynngodau — CHANGES_REQUESTED** [review]
Great to hear the tests are in a better shape than before!

File: plugin-annotation/src/test/java/com/mapbox/mapboxsdk/plugins/annotation/DraggableAnnotationControllerTest.java (hunk 1 above)
Comment (JonasVautherin) [reply plugin-annotation/src/test/java/com/mapbox/mapboxsdk/plugins/annotation/DraggableAnnotationControllerTest.java:268]: Let's see...

Comment (JonasVautherin) [reply plugin-annotation/src/test/java/com/mapbox/mapboxsdk/plugins/annotation/DraggableAnnotationControllerTest.java:268]: Yes you were right! Thanks :pray:

**Review by fynngodau — APPROVED** [review]
Thank you!

**Pull request merged**